### PR TITLE
[JENKINS-36193] Raise exception when trying to build a job that is disabled

### PR DESCRIPTION
**Pull request by godfath3r**
[JENKINS-36193](https://issues.jenkins-ci.org/browse/JENKINS-36193)

### CODE REVIEW

**Comment by jtnord** [comment]
:+1:

**Comment by oleg-nenashev** [comment]
The test failure is unrelated, retriggering just in case.
But 👍

**Comment by oleg-nenashev** [comment]
Retriggering build after https://github.com/jenkinsci/jenkins/pull/2435

**Comment by oleg-nenashev** [comment]
Ready for merge. I'll wait till Friday in the case @godfath3r finds some time to write tests

File: core/src/main/java/jenkins/model/ParameterizedJobMixIn.java [hunk 1]
```diff
@@ -178,7 +178,7 @@ public final void doBuild(StaplerRequest req, StaplerResponse rsp, @QueryParamet
         }
 
         if (!asJob().isBuildable()) {
-            throw HttpResponses.error(SC_INTERNAL_SERVER_ERROR, new IOException(asJob().getFullName() + " is not buildable"));
+            throw HttpResponses.error(SC_CONFLICT, new IOException(asJob().getFullName() + " is not buildable"));
```
Comment (jglick) [inline core/src/main/java/jenkins/model/ParameterizedJobMixIn.java]: Somewhat creative use of a 409, but as good a status as any I suppose.

Comment (jglick) [inline core/src/main/java/jenkins/model/ParameterizedJobMixIn.java]: FWIW `build-token-root` currently [uses a 400](https://github.com/jenkinsci/build-token-root-plugin/blob/1671973b585058b7070b3fce6127613583cb62bb/src/main/java/org/jenkinsci/plugins/build_token_root/BuildRootAction.java#L180).

**Comment by jglick** [comment]
Definitely could use a new test demonstrating the bug. [This](https://github.com/jenkinsci/jenkins/blob/f9efe652490976b03b07a669d3e77dba75b8cce7/test/src/test/java/hudson/model/ParametersTest.java#L212) seems to be the only existing test in Jenkins core covering a related topic, which is alarming.

File: core/src/main/java/jenkins/model/ParameterizedJobMixIn.java (hunk 1 above)
Comment (godfath3r) [inline core/src/main/java/jenkins/model/ParameterizedJobMixIn.java]: > **400 Bad Request** The request could not be understood by the server due to malformed syntax. The client SHOULD NOT repeat the request without modifications.

> **409 Conflict** The request could not be completed due to a conflict with the current state of the resource. This code is only allowed in situations where it is expected that the user might be able to resolve the conflict and resubmit the request. The response body SHOULD include enough

Taken from here: https://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html

409 seems more close to the situation. What do you guys say? If SC_CONFLICT seems better then, build-token-root should also use 409.

**Comment by godfath3r** [comment]
I will try to write some test. Please don't merge until then.

**Comment by oleg-nenashev** [comment]
I'll write the test in a follow-up pull request

**Comment by oleg-nenashev** [comment]
We had a discussion with @godfath3r in IRC about it several days ago

**Comment by oleg-nenashev** [comment]
Merged as a part of https://github.com/jenkinsci/jenkins/pull/2457

**Pull request merged**